File: app/src/main/java/com/lifedawn/bestweather/intro/IntroTransactionFragment.java
package com.lifedawn.bestweather.intro;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.activity.OnBackPressedCallback;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.databinding.FragmentIntroTransactionBinding;
import com.lifedawn.bestweather.main.InitViewModel;
import com.lifedawn.bestweather.main.MyApplication;

import org.jetbrains.annotations.NotNull;

public class IntroTransactionFragment extends Fragment {
	private FragmentIntroTransactionBinding binding;

	private OnBackPressedCallback onBackPressedCallback = new OnBackPressedCallback(true) {
		@Override
		public void handleOnBackPressed() {
			if (!getChildFragmentManager().popBackStackImmediate()) {
				getActivity().finish();
			}
		}
	};

	@Override
	public void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		getActivity().getOnBackPressedDispatcher().addCallback(this, onBackPressedCallback);

		getActivity().getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN |
				View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
	}

	@Nullable
	@org.jetbrains.annotations.Nullable
	@Override
	public View onCreateView(@NonNull @NotNull LayoutInflater inflater, @Nullable @org.jetbrains.annotations.Nullable ViewGroup container,
	                         @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
		binding = FragmentIntroTransactionBinding.inflate(inflater);
		return binding.getRoot();
	}

	@Override
	public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);


	}

	@Override
	public void onStart() {
		super.onStart();
		IntroFragment introFragment = new IntroFragment();
		getChildFragmentManager().beginTransaction().add(binding.fragmentContainer.getId(), introFragment,
				getString(R.string.tag_intro_fragment)).commitNow();
	}

	@Override
	public void onDestroy() {
		super.onDestroy();
		onBackPressedCallback.remove();
	}
}

File: app/src/main/java/com/lifedawn/bestweather/ui/intro/IntroTransactionFragment.kt
package com.lifedawn.bestweather.ui.intro

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.OnBackPressedCallback
import androidx.fragment.app.Fragment
import com.lifedawn.bestweather.R
import com.lifedawn.bestweather.databinding.FragmentIntroTransactionBinding

class IntroTransactionFragment constructor() : Fragment() {
    private var binding: FragmentIntroTransactionBinding? = null
    private val onBackPressedCallback: OnBackPressedCallback = object : OnBackPressedCallback(true) {
        public override fun handleOnBackPressed() {
            if (!getChildFragmentManager().popBackStackImmediate()) {
                getActivity()!!.finish()
            }
        }
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getActivity()!!.getOnBackPressedDispatcher().addCallback(this, onBackPressedCallback)
        getActivity()!!.getWindow().getDecorView().setSystemUiVisibility(
            (View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or
                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR)
        )
    }

    public override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentIntroTransactionBinding.inflate(inflater, container, false)
        return binding!!.getRoot()
    }

    public override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    public override fun onStart() {
        super.onStart()
        val introFragment: IntroFragment = IntroFragment()
        getChildFragmentManager().beginTransaction().add(
            binding!!.fragmentContainer.getId(), introFragment,
            getString(R.string.tag_intro_fragment)
        ).commitNow()
    }

    public override fun onDestroy() {
        super.onDestroy()
        onBackPressedCallback.remove()
    }

    public override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}